### ISSUE TITLE
[LifetimeSafety] False-negative with std::move of pointer types

### ISSUE THREAD

**Issue by usx95** [llvm/llvm-project#191954]
https://godbolt.org/z/rETxncchf
```cpp
int* test2() {
  int x;
  int* f = &x; // Should warn.
  int* a;
  a = std::move(f);
  return a;
}
```

We can special case `std::move` to propagate inner origin list (another use-case for `lifetimebound(2)` for user defined move functions).
